docs/hub/api.md

```diff
@@ -224,4 +230,100 @@ Payload:
 headers = { "authorization" :  "Bearer $token" }
 ```
 
-This is equivalent to `huggingface_hub.whoami()`.
\ No newline at end of file
+This is equivalent to `huggingface_hub.whoami()`.
+
+## Collection API
+
+Use Collections to group repositories from the Hub (Models, Datasets, Spaces and Papers) on a dedicated page.
+
+Find out more about collections [in this guide](./collections.md). Collections can also be managed using the Python client (see [guide](https://huggingface.co/docs/huggingface_hub/main/en/guides/collections)).
+
+### POST /api/collections
+
+Create a new collection on the Hub with a title, a description (optional) and a first item (optional). An item is defined by a type (`model`, `dataset`, `space` or `paper`) and an id (repo_id or paper_id on the Hub).
+
+Payload:
+
+```js
+payload = {
+    "title": "My cool models",
+    "namespace": "username_or_org",
+    "description": "Here is a shortlist of models I've trained.",
+    "item" : {
+        "type": "model",
+        "id": "username/cool-model",
+    }
+    "private": false,
+
+}
+```
+
+This is equivalent to `huggingface_hub.create_collection()`.
+
+### GET /api/collections/{namespace}/{slug}-{id}
+
+Return information about a collection.
+
+This is equivalent to `huggingface_hub.get_collection()`.
+
+### PATCH /api/collections/{namespace}/{slug}-{id}
+
+Update the metadata of a collection on the Hub. You can't add or modify the items of the collection with this method. All fields of the payload are optional.
+
+Payload:
+
+```js
+payload = {
+    "title": "My cool models",
+    "description": "Here is a shortlist of models I've trained.",
+    "private": false,
+    "position": 0, // position of the collection on your profile
+    "theme": "green",
+}
+```
+
+This is equivalent to `huggingface_hub.update_collection_metadata()`.
+### DELETE /api/collections/{namespace}/{slug}-{id}
+
+Return a collection. This is a non-revertible operation. A deleted collection cannot be restored.
+
+This is equivalent to `huggingface_hub.delete_collection()`.
+
+### POST /api/collections/{namespace}/{slug}-{id}/items
```

Comment:
those endpoints don't accept just `{id}`, they also require the slug?

Reply:
Not at the moment no.

You can do:
- https://huggingface.co/api/collections/TheBloke/recent-models-64f9a55bb3115b4f513ec026
- https://huggingface.co/api/collections/TheBloke/abcdef-64f9a55bb3115b4f513ec026

But not:
- https://huggingface.co/api/collections/TheBloke/-64f9a55bb3115b4f513ec026
- https://huggingface.co/api/collections/TheBloke/64f9a55bb3115b4f513ec026

Reply:
maybe we can implem that server side (in a future PR) if it improves DX significantly (maybe not worth it?)

Reply:
Not sure it'll help much. In any case the user will have to copy paste something right? (and copy-pasting `username/slug-id` from a URL is easier than copy-pasting `username` and `id` separately). And if not copy-pasting then it'll be programmatically done and in that case the server returns the full version (`username/slug-id`).

Reply:
good point about the copy-pasting